src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.cs

```diff
@@ -383,5 +246,603 @@ public static bool TryGetFloatingPointConstant(ReadOnlySpan<byte> span, out doub
             value = 0;
             return false;
         }
+
+#if PLATFORM_INTRINSICS
+        // Borrowed and modified from SpanHelpers.Byte:
+        // https://github.com/dotnet/runtime/blob/a196d534495a1654ffd158211ca8761b1eba8c05/src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs#L916-L1207
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        private static int IndexOfOrLessThan(ref byte searchSpace, byte value0, byte value1, byte lessThan, int length)
+        {
+            Debug.Assert(length >= 0);
+
+            uint uValue0 = value0; // Use uint for comparisons to avoid unnecessary 8->32 extensions
+            uint uValue1 = value1; // Use uint for comparisons to avoid unnecessary 8->32 extensions
+            uint uLessThan = lessThan; // Use uint for comparisons to avoid unnecessary 8->32 extensions
+            nuint offset = 0; // Use nuint for arithmetic to avoid unnecessary 64->32->64 truncations
+            nuint lengthToExamine = (nuint)(uint)length;
+
+            if (Sse2.IsSupported || AdvSimd.Arm64.IsSupported)
```

Comment:
How much additional gain does this provide over just the `Vector<T>` path?

Reply:
Couldn't say, however is essentially syncing to the code to the three value `SpanHelpers.Byte.IndexOfAny`; but switching the last test for less than rather than equals.

Reply:
Sorry, there were results :)

Up to +20% improvement as `Vector<T>` only covers longest length, whereas this covers multiple vector lengths